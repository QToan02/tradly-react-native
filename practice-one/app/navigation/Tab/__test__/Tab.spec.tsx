import { render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'

import BottomNav from '../index'

describe('Testing bottom tab navigation', () => {
  it('Render tab successfully', () => {
    const component = render(
      <NavigationContainer>
        <BottomNav />
      </NavigationContainer>
    ).toJSON()

    expect(component).toMatchSnapshot()
  })
})
