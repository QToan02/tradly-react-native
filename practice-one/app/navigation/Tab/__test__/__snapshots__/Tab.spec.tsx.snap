// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing bottom tab navigation Render tab successfully 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "paddingHorizontal": 16,
                  "paddingVertical": 22,
                },
                {
                  "backgroundColor": "#33907C",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                  {
                    "marginBottom": 14,
                  },
                  {
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#fff",
                      "fontFamily": "Montserrat_700Bold",
                      "fontSize": 24,
                    },
                    false,
                    {},
                  ]
                }
              >
                Home
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                    "marginVertical": 14,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 60,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 5"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": true,
          }
        }
        accessibilityStates={
          [
            "selected",
          ]
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            {
              "marginVertical": 13,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/home.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#33907C",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/home.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#4f4f4f",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#33907C",
              },
              {
                "fontSize": 10,
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "fontSize": 10,
              },
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Browse, tab, 2 of 5"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            {
              "marginVertical": 13,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/search.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#33907C",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/search.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#4f4f4f",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "fontSize": 10,
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "fontSize": 10,
              },
            ]
          }
        >
          Browse
        </Text>
      </View>
      <View
        accessibilityLabel="Product, tab, 3 of 5"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            {
              "marginVertical": 13,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/store.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#33907C",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/store.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#4f4f4f",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "fontSize": 10,
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "fontSize": 10,
              },
            ]
          }
        >
          Product
        </Text>
      </View>
      <View
        accessibilityLabel="Order History, tab, 4 of 5"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            {
              "marginVertical": 13,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/order.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#33907C",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/order.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#4f4f4f",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "fontSize": 10,
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "fontSize": 10,
              },
            ]
          }
        >
          Order History
        </Text>
      </View>
      <View
        accessibilityLabel="Profile, tab, 5 of 5"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            {
              "marginVertical": 13,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/profile.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#33907C",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/tab/profile.png",
                }
              }
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "tintColor": "#4f4f4f",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "fontSize": 10,
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "fontSize": 10,
              },
            ]
          }
        >
          Profile
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
